javascript/ajax.js

```diff
@@ -881,69 +881,82 @@ function buildSendParams(obj) {
 }
 
 function update_db(obj) {
-    var callback = obj.data.callback;
-    var url = obj.data.script;
-    var fetch_url = obj.data.fetch_url;
-    var analysis_id = obj.data.analysis_id;
-    var payload = buildURL(this);
-    if (!payload) {return};
-    $.ajax({url        : url,
-	    type       : "post",
-	    data       : payload,
-	    dataType   : "json",
-	    async      : false,
-	    cache      : false,
-	    success    : function(updateRes) {
-		if(updateRes.status !== "ok") {
-		    log(updateRes);
-		};
-	    },
-	    // TODO: I think the log is populated twice... One in the success callback and one in the
-	    // onSuccess_fetchAnalysis callback. Check!
-	    complete   : function() { display(analysis_id, fetch_url, callback)}
-	   });
+    buildURL(this, actuallyUpdateDb);
+
+    function actuallyUpdateDb(sqlLocator) {
+        var callback = obj.data.callback;
+        var url = obj.data.script;
+        var fetch_url = obj.data.fetch_url;
+        var analysis_id = obj.data.analysis_id;
+
+        $.ajax({url        : url,
+            type       : "post",
+            data       : sqlLocator,
+            dataType   : "json",
+            async      : false,
```

Comment:
This field is almost certainly unnecessary (and, according to the [docs](https://api.jquery.com/jQuery.ajax/#jQuery-ajax-settings), deprecated for the jQuery version that is used in this project)